### PR TITLE
Added Category model

### PR DESCRIPTION
**Pull request by ynik404**

### CODE REVIEW

File: src/main/java/mate/academy/dto/book/BookDto.java [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
+import java.util.Set;
 import lombok.Data;
 
 @Data
 public class BookDto {
-    private Long id;
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/book/BookDto.java:11]: NotBlank for String

File: src/main/java/mate/academy/model/Category.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(unique = true, nullable = false)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    private String description;
+    @Column(nullable = false)
+    private Boolean isDeleted = false;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Category.java:29]:
```suggestion
    private boolean isDeleted = false;
```

File: src/main/java/mate/academy/controller/BookController.java [hunk 3]
```diff
@@ -24,56 +23,63 @@
...
-    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PreAuthorize("hasRole('ADMIN')")
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    @Operation(summary = "Create a new book", description = "Create a new book")
-    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
-        return bookService.createBook(requestDto);
+    @Operation(summary = "Create a new book",
+            description = "Create a new book")
+    public BookDto createBook(@RequestBody @Valid BookDto bookDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/BookController.java:55]: Why did you change it? 
You should return BookDto with Id and provide request Dto without

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 4]
```diff
@@ -18,9 +35,10 @@ databaseChangeLog:
                   type: varchar(255)
                   constraints:
                     nullable: false
+                    unique: true
               - column:
                   name: password
-                  type: varchar(255)
+                  type: varchar(2000)
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml:41]: why did you decide to change it?

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 5]
```diff
@@ -35,10 +53,43 @@ databaseChangeLog:
                     nullable: false
               - column:
                   name: shipping_address
-                  type: varchar(255)
+                  type: varchar(500)
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml:56]: why did you decide to change it?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 6]
```diff
@@ -42,3 +60,36 @@ databaseChangeLog:
...
+            constraintName: fk_user_id
+            referencedTableName: users
+            referencedColumnNames: id
+
+        - addForeignKeyConstraint:
+            baseTableName: users_roles
+            baseColumnNames: role_id
+            constraintName: fk_role_id
+            referencedTableName: roles
+            referencedColumnNames: id
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml:95]: please always add a new empty line at the end when you this red symbol
image

File: src/main/java/mate/academy/service/impl/CategoryServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                .orElseThrow(() -> new EntityNotFoundException("Can't find category by id: " + id));
+        categoryMapper.updateCategoryFromDto(requestDto, category);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
```
Comment (okuzan) [inline src/main/java/mate/academy/service/impl/CategoryServiceImpl.java:57]:
```suggestion
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**